### PR TITLE
(PC-7614) fix incorrect password behavior

### PR DESCRIPTION
**Pull request by emmajmn**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-7614

## Checklist

I have:

- [ ] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [ ] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/profile/pages/ChangePassword.tsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ export function ChangePassword() {
   // on type, reset error field
   useEffect(() => {
     if (hasError) {
-      // setHasError(false)
+      setHasError(false)
```
Comment (antoinewg) [inline src/features/profile/pages/ChangePassword.tsx:44]: ce serait intéressant de comprendre pk ce bout de code a été commenté. L'origine du bug

Comment (antoinewg) [reply src/features/profile/pages/ChangePassword.tsx:44]: encore une fois, vu que tu corriges une régression, ce serait cool de rajouter un test.
Là c'est un peu bcp de demander vu que t'es juste là pour réparer le bout de code, mais on sera toujours plus serein.

Comment (emmajmn) [reply src/features/profile/pages/ChangePassword.tsx:44]: yep je regarde pour ajouter un test !

**Review by antoinewg — APPROVED** [review]
cool merci 👍

**Pull request merged**